packages/client/src/features/grid/grid-background.ts

```diff
@@ -0,0 +1,72 @@
+/********************************************************************************
+ * Copyright (c) 2024 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import {
+    Action,
+    IActionHandler,
+    MaybePromise,
+    Point,
+    SetViewportAction,
+    TYPES,
+    ViewerOptions,
+    Viewport,
+    findParentByFeature,
+    isViewport
+} from '@eclipse-glsp/sprotty';
+import { inject, injectable } from 'inversify';
+import { EditorContextService, IDiagramStartup } from '../../base';
+import { Grid } from './grid';
+
+@injectable()
+export class GridBackground implements IActionHandler, IDiagramStartup {
```

Comment:
Is there a particular reason why the grid is not rendered as part of the graph view?
Using a service here that is separated from our default entrypoints to manipulate the ui (views, uiextensions, vnode postprocessors) seems a bit of to me.

It creates a lot of overhead like dedicated feedback action and command, actions handlers etc. just to handle something that can be handled a lot easier in the graph view.

In addition, the grid rendering is currently off if I use the center/fit-to-screen actions.  Until the next viewport change the grid is not rendered correctly

Reply:
That is a very interesting idea that didn't occur to me first. I had a closer look at it and it is indeed a better approach since we do not have to care about how the viewport is manipulated. Fantastic idea, thank you! I adapted the code to render the background as part of the graph. Thanks!